src/main/java/swinglib/JButtonModal.java

```diff
@@ -0,0 +1,70 @@
+package swinglib;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import javax.swing.JButton;
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+
+public class JButtonModal {
+
+  private final String title;
+  private final String toolTip;
+  private final Runnable actionListener;
+
+  public JButtonModal(final String title, final String toolTip, final Runnable actionListener) {
+    this.title = Preconditions.checkNotNull(title);
+    this.toolTip = toolTip;
+    this.actionListener = Preconditions.checkNotNull(actionListener);
+  }
+
+  public static JButtonModalBuilder builder() {
+    return new JButtonModalBuilder();
+  }
+
+  private JButton swingComponent() {
+    final JButton button = new JButton(title);
+    if (toolTip != null) {
+      button.setToolTipText(toolTip);
+    }
+    button.addActionListener(e -> actionListener.run());
+    return button;
+  }
+
+
+  public static class JButtonModalBuilder {
```

Comment:
Great suggestion, thank you. I'll incorporate it in. It's pretty slick that the redundant variables were eliminated.